internal/controller/util/vm_util.go

```diff
@@ -57,35 +67,267 @@ func ListVMsByVMNamespace(
 	log logr.Logger,
 	vmNamespaceList []string,
 	vmList []string,
-) ([]string, error) {
-	var foundVMList []string
-
-	var notFoundErr error
-
-	foundVM := &virtv1.VirtualMachine{}
+) ([]virtv1.VirtualMachine, error) {
+	var foundVMs []virtv1.VirtualMachine
+	// Set to track seen (ns, name) pairs
+	seen := make(map[string]struct{}, len(vmNamespaceList)*len(vmList))
```

Comment:
I am not sure why we need `seen`, is it to deal with a repeated namespace in the vmNamespaceList input parameter? Also this can hold a bool, why a struct{} as the value?